k2/csrc/fsa_algo.h

```diff
@@ -298,6 +298,38 @@ Ragged<int32_t> ShortestPath(FsaVec &fsas,
  */
 Fsa Union(FsaVec &fsas, Array1<int32_t> *arc_map = nullptr);
 
+/* Compute the closure of an FSA.
+
+   The closure is implemented in the following way:
+
+   - For all arcs entering the final-state, change the destination-state from
+   the final-state to state 0 (the start state), and the label from -1 to 0.
+
+   - Add an arc from the start-state to the final-state (with label -1, and
+   score 0, of course); this can be the last numbered arc of the start state of
+   the output. The implementation wouldn't be in-place, we'd create a new FSA.
+   Assume that the original start state has two arcs, then the arc-map would be
+   [ 0 1 -1 2 3 ... ].
+
+   Note: the output will be arc-sorted if the input was arc-sorted and
+   epsilon-free (actually we only need for the start-state to not have epsilon
+   arcs leaving it). The output will not be epsilon-free though, so if an
+   epsilon-free output is desired, which it generally will be, the caller will
+   need to determinize afterward.
```

Comment:
`the caller should call rmepsilon afterward`?

Reply:
Some states of the returned FSA  may have more than two leaving arcs with label == 0 even if
the input FSA is determinized.

Reply:
> The output will not be epsilon-free though, so if an epsilon-free output is desired, which it generally will be, the caller will need to determinize afterward.

The point here is even we call determinize, the output will still have epsilon-arcs (even though the input is epsilon-free), as we have changed label from -1 to 0 for those final-arcs. so `if an epsilon-free output is desired` , we should call rm-epsilon instead of determinize.

BTW,  there may be two arcs with lable -1 from start_state to final_state in current approach.

Reply:
Yes, you're right. Will change it.